command/debug.go

```diff
@@ -731,35 +731,35 @@ func (c *DebugCommand) collectPprof(ctx context.Context) {
 
 		var wg sync.WaitGroup
 
-		// Capture goroutines
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			data, err := pprofGoroutine(ctx, c.cachedClient)
-			if err != nil {
-				c.captureError("pprof.goroutine", err)
-				return
-			}
+		for _, target := range []string{"threadcreate", "allocs", "block", "mutex", "goroutine", "heap"} {
+			wg.Add(1)
+			go func(target string) {
+				defer wg.Done()
+				data, err := pprofTarget(ctx, c.cachedClient, target, nil)
+				if err != nil {
+					c.captureError("pprof."+target, err)
+					return
+				}
 
-			err = ioutil.WriteFile(filepath.Join(dirName, "goroutine.prof"), data, 0o644)
-			if err != nil {
-				c.captureError("pprof.goroutine", err)
-			}
-		}()
+				err = ioutil.WriteFile(filepath.Join(dirName, target+".prof"), data, 0o644)
+				if err != nil {
+					c.captureError("pprof."+target, err)
+				}
+			}(target)
+		}
 
-		// Capture heap
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			data, err := pprofHeap(ctx, c.cachedClient)
+			data, err := pprofTarget(ctx, c.cachedClient, "goroutine", url.Values{"debug": []string{"2"}})
 			if err != nil {
-				c.captureError("pprof.heap", err)
+				c.captureError("pprof.goroutines-text", err)
```

Comment:
Out of curiosity, why do we have the `-text` suffix here?

Reply:
To distinguish it from the `pprof.goroutine` target.